virttest/utils_vdpa.py

```diff
@@ -0,0 +1,251 @@
+"""
+Virtualization test - vDPA related utilities
+
+:copyright: Red Hat Inc.
+"""
+import logging
+import time
+
+from avocado.core import exceptions
+from avocado.utils import process
+
+from virttest import openvswitch
+from virttest import utils_misc
+from virttest import utils_sriov
+from virttest import utils_switchdev
+from virttest.utils_kernel_module import KernelModuleHandler
+from virttest.versionable_class import factory
+
+LOG = logging.getLogger('avocado.' + __name__)
+
+
+class VDPAOvsTest(object):
+    """
+    Wrapper class for vDPA OVS environment configuration
+    """
+    def __init__(self, pf_pci, vf_no=4):
+        self.pf_pci = pf_pci
+        self.vf_no = vf_no
+        self.pf_pci_path = utils_misc.get_pci_path(self.pf_pci)
+        utils_sriov.set_vf(self.pf_pci_path, 0)
+
+        self.pf_iface = utils_sriov.get_pf_info_by_pci(self.pf_pci).get('iface')
+        if not self.pf_iface:
+            raise exceptions.TestCancel("NO available pf found.")
+        self.br_name = self.pf_iface+'_br'
+
+        self.ovs = factory(openvswitch.OpenVSwitchSystem)()
+
+    def __del__(self):
+        self.cleanup()
+
+    def load_modules(self):
+        """
+        Load modules
+        """
+        modules = ['vdpa', 'vhost_vdpa', 'mlx5_vdpa']
+        for module_name in modules:
+            KernelModuleHandler(module_name).reload_module(True)
+
+    def unload_modules(self):
+        """
+        Unload modules
+        """
+        modules = ['mlx5_vdpa', 'vhost_vdpa', 'vdpa']
+        for module_name in modules:
+            KernelModuleHandler(module_name).unload_module()
+
+    def config_vdpa_dev(self):
+        """
+        Bind VFs and add vDPA dev
+        """
+        utils_switchdev.bind_vfs(self.pf_pci, self.vf_no)
+        for idx in range(self.vf_no):
+            vf_pci = utils_sriov.get_vf_pci_id(self.pf_pci, vf_index=idx)
+            vf_dev = utils_sriov.get_iface_name(vf_pci)
+            if not vf_dev:
+                raise exceptions.TestError("Cannot get VF network device!")
+            self.set_tc_offload(vf_dev)
+            self.set_dev_manged_no(vf_dev)
+            self.add_vdpa_dev(idx, vf_pci)
+
+    def get_rep_list(self):
+        """
+        Get representor list
+        """
+        return utils_switchdev.get_rep_list(self.pf_iface, self.vf_no)
+
+    def config_reps(self):
+        """
+        Configure reps
+        """
+        for rep in self.get_rep_list():
+            self.set_tc_offload(rep)
+            self.set_dev_manged_no(rep)
+            self.set_ifc_up(rep)
+
+    def add_vdpa_dev(self, idx, pci_addr):
+        """
+        Add vDPA device
+
+        :param idx: Index of dev
+        :param pci_addr: PCI address
+        """
+        cmd = "vdpa dev add name vdpa{} mgmtdev pci/{}".format(idx, pci_addr)
+        process.run(cmd, shell=True)
+
+    def set_dev_manged_no(self, dev):
```

Comment:
Would be better to cal function set_dev_managed_no, what do you think?

Reply:
Oops! Updated. Thank you!